index/scorch/segment/int.go

```diff
@@ -92,3 +95,84 @@ func DecodeUvarintAscending(b []byte) ([]byte, uint64, error) {
 	}
 	return b[length:], v, nil
 }
+
+// ------------------------------------------------------------
+
+type MemUvarintReader struct {
+	C int64 // index of next byte to read from S
```

Comment:
just curious to know int vs int64 for the index position? former helps to save some casting.. :)

Reply:
yah, the int64 stuff came from the original bytes.Reader approach -- i assumed the compiler just made casts like this automagically no-op away on the right platforms, but should check

Reply:
thanks!  looks like going with int's and avoiding the casting was helpful, so I've just pushed up another commit to the end of this pull-request chain on that